### PR TITLE
Add missing test cases to bump up coverage for Plank

### PR DESCRIPTION
**Pull request by jooskim**
This PR adds some unit tests and integration test cases for critical paths covering usual and some edge case scenarios to ensure Plank works properly in various conditions. Also by writing tests I was able to catch some minor/moderate issues and correct them in the meantime.

### CODE REVIEW

File: plank/pkg/server/server_test_utils.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+func runWhenServerReady(t *testing.T, eventBus bus.EventBus, fn func(t *testing.T)) {
+	handler, _ := eventBus.ListenOnce(PLANK_SERVER_ONLINE_CHANNEL)
+	handler.Handle(func(message *model.Message) {
+		fn(t)
+	}, func(err error) {
+		assert.FailNow(t, err.Error())
+	})
+}
+
+func getTestPort() int {
```
Comment (daveshanley) [inline plank/pkg/server/server_test_utils.go:58]: You should add this to the  `test_utils` directory and make it public.

File: plank/pkg/server/server_test_utils.go [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+	protocol := "http"
+	if config.TLSCertConfig != nil {
+		protocol += "s"
+	}
+
+	baseUrl = fmt.Sprintf("%s://%s:%d", protocol, config.Host, config.Port)
+	return baseUrl, config.LogConfig.OutputLog, testServer
+}
+
+func runWhenServerReady(t *testing.T, eventBus bus.EventBus, fn func(t *testing.T)) {
```
Comment (daveshanley) [inline plank/pkg/server/server_test_utils.go:49]: You should add this to the test_utils directory and make it public.

File: plank/pkg/server/server_test_utils.go [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+			ErrorLog:      errLog,
+			FormatOptions: &utils.LogFormatOption{},
+		},
+		NoBanner:                 noBanner,
+		ShutdownTimeoutInMinutes: 1,
+	}
+	return cfg
+}
+
+func createTestServer(config *PlatformServerConfig) (baseUrl, logFile string, s PlatformServer) {
```
Comment (daveshanley) [inline plank/pkg/server/server_test_utils.go:37]: You should add this to the test_utils directory and make it public.

File: plank/pkg/server/server_test_utils.go [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+	"net"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+)
+
+var testSuitePortMap = make(map[string]int)
+
+func getBasicTestServerConfig(rootDir, outLog, accessLog, errLog string, port int, noBanner bool) *PlatformServerConfig {
```
Comment (daveshanley) [inline plank/pkg/server/server_test_utils.go:19]: use the public one in `test_utils`

File: plank/pkg/server/server_test_utils.go (hunk 1 above)
Comment (jooskim) [reply plank/pkg/server/server_test_utils.go]: actually this causes an import cycle between `server` and `test_utils` packages because all the test cases are within the `server` package/folder and so it forms a circular reference of the two packages. i am afraid we might as well put the test harness stuff into the `server` package because i don't see a clean way to keep them in another package while preventing import cycles. let me know what you think

Comment (daveshanley) [reply plank/pkg/server/server_test_utils.go]: OK, leave it as is. makes sence.

**Review by daveshanley — APPROVED** [review]
LGTM
